config-store-k8s-config-map/src/main/java/io/stargate/config/store/k8s/config/map/ConfigStoreK8sConfigMap.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.config.store.k8s.config.map;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.DoneableConfigMap;
+import io.fabric8.kubernetes.client.NamespacedKubernetesClient;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.stargate.config.store.ConfigStore;
+import java.io.IOException;
+import java.util.Map;
+
+public class ConfigStoreK8sConfigMap implements ConfigStore {
+  private final NamespacedKubernetesClient client;
+  private final String namespace;
+
+  public ConfigStoreK8sConfigMap(String namespace, NamespacedKubernetesClient client) {
+    this.namespace = namespace;
+    this.client = client;
+  }
+
+  @Override
+  public Map<String, String> getConfigForExtension(String extensionName) {
+    Resource<ConfigMap, DoneableConfigMap> configMaps =
+        client.configMaps().inNamespace(namespace).withName(extensionName);
+    return configMaps.get().getData();
+  }
+
+  @Override
+  public void addOrReplaceConfigValueForExtension(String extensionName, String key, String value) {
+    ConfigMap configMap = client.configMaps().inNamespace(namespace).withName(extensionName).get();
+    if (configMap == null) {
+      // create new config map
+      ConfigMap newConfigMap =
+          new ConfigMapBuilder()
+              .withNewMetadata()
+              .withName(extensionName)
+              .endMetadata()
+              .addToData(key, value)
+              .build();
+      client.configMaps().inNamespace(namespace).create(newConfigMap);
+    } else {
+      // create a config map based on the existing one
+      ConfigMap newConfigMap =
+          new ConfigMapBuilder()
+              .withNewMetadata()
+              .withName(extensionName)
+              .endMetadata()
+              .addToData(configMap.getData())
```

Comment:
We may have a race condition here. One node may see `configMap == null` whereas others may see already created `configMap.` But I think that this is not a problem - the add to data operation is idempotent because it does not keep duplicates. Duplicate overrides the previous value, so the last one wins.
The 2nd case is more problematic - if one node already creates a configMap, the other will create a new empty map overriding previous values. I am not sure if k8s has a pattern for doing that with the `config-map` API.

Reply:
One issue with config maps is that they are eventually consistent. So if I were to create a config map and bind it to 3 pods then they're not all going to see the change immediately but as they poll each of them will see the file based on their polling.